include/dlaf/common/bag.h

```diff
@@ -0,0 +1,70 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2021, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <hpx/tuple.hpp>
+
+#include "dlaf/common/callable_object.h"
+#include "dlaf/common/data_descriptor.h"
+#include "dlaf/matrix/tile.h"
+
+/// @file
+
+namespace dlaf {
+namespace common {
+
+namespace internal {
+
+/// A Bag:
+/// - owns the temporary buffer (optionally allocated)
+/// - contains a data descriptor to the contiguous data to be used for communication
+///   (being it the aforementioned temporary buffer if allocated, otherwise an externally managed memory,
+///   e.g. a tile)
+///
+/// It is up to the user to guarantee the integrity of the Bag and its properties
+template <class T>
+using Bag = hpx::tuple<common::Buffer<std::remove_const_t<T>>, common::DataDescriptor<T>>;
```

Comment:
Might this also benefit from being a struct with named fields rather than a tuple? Or did I miss a case where you're using this with `split_future`?

Reply:
I really don't want to block this on the name so let's keep `Bag` if we don't have a better alternative. But maybe `ContiguousBufferHolder`? Not a great name either but `Bag` is *very* generic ;)

Reply:
Something like this? do you have any better suggestion for the field names?

```cpp
template <class T>
struct ContiguousBufferHolder {
  common::Buffer<std::remove_const_t<T>> buffer;
  common::DataDescriptor<T> descriptor;
};
```

Reply:
No better suggestion, this already looks much better! Thanks.